cloud/hack/test.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_RACE=${KUBE_RACE:-}
+eval "testargs=(${KUBE_TEST_ARGS:-})"
+SCRIPT_ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/../../ && pwd)
+
+
+findTestDirs() {
+  (
+    cd ${SCRIPT_ROOT}/cloud
+    find -L . -not \( \
+        \( \
+          -path './test/integration/*' \
+        \) -prune \
+      \) -name '*_test.go' -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
+  )
+}
+
+
+testcases=()
+for arg; do
+  if [[ "${arg}" == -* ]]; then
+    goflags+=("${arg}")
+  else
+    testcases+=("${arg}")
```

Comment:
where's the testcases used when runTests

Reply:
testcases are used to run unit tests.  When the integration tests are run , these are not useful.
however , when unit tests need to be run , the correct test directories are found and set in testcases variable.  this variable is then set as an argument on line 49.